mcs/class/corlib/System.Runtime.Remoting.Proxies/RealProxy.cs

```diff
@@ -49,8 +49,45 @@ namespace System.Runtime.Remoting.Proxies
 	[StructLayout (LayoutKind.Sequential)]
 	internal class TransparentProxy {
 		public RealProxy _rp;
-		IntPtr _class;
+		unsafe Mono.RuntimeStructs.RemoteClass* _class;
```

Comment:
A `RemoteClass` is not a `MonoType` (aka `RuntimeTypeHandle`), actually it's not even a `MonoClass`, it's a separate ad-hoc structure.  I can wrap it up in a value type like `RuntimeTypeHandle` wraps up `MonoType*`, though.